map2.js

```diff
@@ -31,3 +31,30 @@ var phoneBook = [
 // For example, the 0th element of the new array should be {Abe: "1-111-111-1111"}
 
 //Your code here
+
+//this didnt work
+// for(var person in phoneBook){
+//   if(phoneBook[person] === "333-333-3333"){
+//    phoneBook.unshift(3);
+//   }
+// }
+
+// console.log(phoneBook);
+
+
+const newList= phoneBook.map(x=> x+1);
+
+console.log(newList);
+//confused with this one/
+
+
+// for(var person in phoneBook){
+//   if(phoneBook[person] === "333-333-3333"){
+//     console.log(person)
+//   }
+// }
+
+// var arrOfObj = [{name: 'eve'},{name:'john'},{name:'jane'}];
+// arrOfObj.map(function (obj) { 
+//    obj.isActive = true;
+// });
```

Comment:
yeah, this one took a bit more work than the others. the goal is to add a country code to each phone number, where the country code is the same digit as the rest of the digits in the phone number.

a couple of things to consider if you try to re-do this:
* your `map()` is going to take each of the objects individually. In order to change the phone number, you're going to have to get the key inside the object. And since each object has a different key (based on the person's name), you're going to have to get the keys in the individual object programmatically. I would search MDN for a way to get all the keys in an object, if you don't remember how
* Keep in mind that both arrays and strings have index values. `someArray[0]` will return the first element in `someArray`, and `someString[0]` will return the first **character** in `someString`. Use this to get the first character in the phone number, and to add the same character to the beginning of the phone number. Between that and string concatenation or string interpolation, `111-111-1111` should become `1-111-111-1111`. 

If you have time, try this out with the hints I've provided. Also feel free to bring this up during office hours!